### PR TITLE
In LocationContext, tag stop sequences by their routes

### PR DESCRIPTION
**Pull request by jzimbel-mbta**
**Asana task**: ad hoc

This stores some extra data in the `LocationContext` struct, which is useful for disruption diagrams for now, and maybe other things in the future.

Code expecting the original, untagged list of stop sequences can now get it by calling `LocationContext.stop_sequences(location_context)` if it's already inside the struct, or `RoutePattern.untag_stop_sequences(tagged_stop_sequences)` if handling the tagged sequence map directly.

- [x] Tests added? <sup>(None added, but many updated)</sup>

### CODE REVIEW

**Review by hannahpurcell — COMMENTED** [review]
Ok great! Only one comment on... naming 😅

File: lib/screens/v2/location_context.ex
Comment (jzimbel-mbta) [reply lib/screens/v2/location_context.ex:20]: Omg what was I thinking with that name. `stop_sequences_by_route` is much more standard for a map like that.

I already pulled in this commit and added a bunch of commits on top of it in the disruption diagram branch, though—would it be ok if I stick with the current name and submit a separate PR to rename it after the disruption diagram work is merged into `prefare-alerts`?

**Review by hannahpurcell — APPROVED** [review]
Approving with the plan to rename the `tagged` field once the code is merged to `prefare-alerts`

**Pull request merged**
